RecipeBook/ViewControllers/NameScreenViewController/NameScreenViewController.swift

```diff
@@ -19,11 +20,28 @@ class NameScreenViewController: UIViewController {
         nameTextField.delegate = self
     }
 
+    override func viewDidAppear(_ animated: Bool) {
+        let launchType = UserDefaultsService()
+        if !launchType.isFirstLaunch() {
+            let storyboard = UIStoryboard(name: "HomeScreenViewController", bundle: nil)
+            let homeScreenViewController =
+            storyboard.instantiateViewController(withIdentifier: "HomeScreenViewController")
+            homeScreenViewController.modalPresentationStyle = .fullScreen
+            present(homeScreenViewController, animated: false)
+        }
+    }
+
 }
 
 extension NameScreenViewController: UITextFieldDelegate {
 
     func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        // Save User Credentials (Name)
+        let userDefaultsName = UserDefaultsService()
```

Comment:
Create a view model for this class and declare `User Defaults` as a property of that class. Create methods for getting and setting the values in the view model and call them from there.